File: core/src/main/java/com/serious/service/ServiceException.java
package com.serious.service;/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

/**
 * @author Andreas Ernst
 */
public class ServiceException extends RuntimeException {
    // constructor

    public ServiceException(String message) {
        super(message);
    }

    ServiceException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: core/src/main/java/com/serious/service/ServiceException.kt
package com.serious.service
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

/**
 * @author Andreas Ernst
 */
class ServiceException : RuntimeException {
    // constructor
    constructor(message: String?) : super(message)
    internal constructor(message: String?, cause: Throwable?) : super(message, cause)
}
